src/test/java/guitests/UITest.java

```diff
@@ -375,14 +379,58 @@ public void clickMenu(String... menuNames) {
     public <T> void waitForValue(ComboBoxBase<T> comboBoxBase) {
         waitUntil(comboBoxBase, c -> c.getValue() != null);
     }
-    
+
+    /**
+     * Traverses HubTurbo's menu, looking for a chain of nodes with the
+     * given names and triggering their associated action.
+     *
+     * This is a more reliable method of triggering menu items than
+     * {@link #clickMenu}, especially when dealing with nested menu items.
+     * It is a drop-in replacement in most cases.
+     *
+     * Caveats: ensure that adequate synchronisation is used after this method
+     * if it is called from a thread other than the UI thread.
+     *
+     * @param names the chain of menu nodes to visit
+     */
+    public void traverseMenu(String... names) {
+        assert names.length > 0 : "traverseMenu called with no arguments";
+
+        Platform.runLater(() -> {
+            MenuControl root = TestController.getUI().getMenuControl();
+            MenuItem current = root.getMenus().stream()
+                .filter(m -> m.getText().equals(names[0]))
+                .findFirst()
+                .orElseThrow(() ->
+                    new IllegalArgumentException(
+                        String.format("%s is not a valid menu item", names[0])));
+
+            for (int i = 1; i < names.length; i++) {
+                final int j = i;
+                if (current instanceof Menu) {
+                    current = ((Menu) current).getItems().stream()
+                        .filter(m -> m.getText().equals(names[j]))
+                        .findFirst()
+                        .orElseThrow(() ->
+                            new IllegalArgumentException(
+                                String.format("%s is not a valid menu item", names[j])));
+                } else {
+                    throw new IllegalArgumentException(
+                        String.format("Menu %s is not as nested as arguments require", names[0]));
+                }
```

Comment:
We can use `if (!(current instanceof Menu)`)` and move the `throw` statement up so we can eliminate the `else` clause.